### PR TITLE
Prevent calculate percentage directive setting percentageComplete to infinity.

### PR DESCRIPTION
**Pull request by Minglee01**
@todd-dembrey 
@maxpeterson 

please review
thanks

### CODE REVIEW

**Comment by todd-dembrey** [comment]
Would it have been better to ensure that the calculation can handle zero values? You are separating the logic otherwise.

Also would make it easy to write a few tests :+1:

https://github.com/incuna/incuna-surveys/blob/1698b640f8f9d7ce2e4fa70b40246e0914cc61c3/web/app/scripts/src/services/calculate-percentage.es.js#L53

File: web/app/scripts/src/directives/calculate-percentage.es.js [hunk 1]
```diff
@@ -29,10 +29,12 @@ module.directive('calculatePercentage', [
                 });
 
                 // Using true to compare the sub-elements.
-                scope.$watch('model', (answers) => {
-                    const numberOfCompletedQuestions = CalculateCompletion.countNumberOfAnsweredQuestions(answers);
-                    scope.percentageComplete = CalculateCompletion.calculatePercentageComplete(numberOfCompletedQuestions, totalQuestionCount);
-                }, true);
+                if (totalQuestionCount > 0) {
```
Comment (todd-dembrey) [inline web/app/scripts/src/directives/calculate-percentage.es.js]: Should this be reverted now?

Comment (Minglee01) [inline web/app/scripts/src/directives/calculate-percentage.es.js]: we dont want to trigger that watch. untill we have a question count

Comment (Minglee01) [inline web/app/scripts/src/directives/calculate-percentage.es.js]: cause a watch makes the app slower

Comment (maxpeterson) [inline web/app/scripts/src/directives/calculate-percentage.es.js]: The `link` is only run once (per directive on the page) so I would expect `totalQuestionCount` to always be `0` during the `link` function. I think you need to put the optimisation in the watch:
```
                scope.$watch('model', (answers) => {
                    if (totalQuestionCount == 0) {
                      return;
                    }
                    const numberOfCompletedQuestions = CalculateCompletion.countNumberOfAnsweredQuestions(answers);
                    scope.percentageComplete = CalculateCompletion.calculatePercentageComplete(numberOfCompletedQuestions, totalQuestionCount);
                }, true);
```

File: web/tests/calculate-percentage-directive-spec.js [hunk 2]
```diff
@@ -45,20 +42,17 @@ describe('calculatePercentage directive', function () {
             this.elm = this.$compile(tpl)(this.scope);
             this.scope.$digest();
         };
-        spyOn(this.calculateCompletionPercent, 'countQuestionsTotal').and.returnValue(questionSet);
-        spyOn(this.calculateCompletionPercent, 'countNumberOfAnsweredQuestions').and.returnValue(answers);
     });
 
-    it('should call this.calculateCompletionPercent.countQuestionsTotal with the questionSet', function () {
```
Comment (maxpeterson) [inline web/tests/calculate-percentage-directive-spec.js]: What has happened to this test?

File: web/app/scripts/src/directives/calculate-percentage.es.js (hunk 1 above)
Comment (Minglee01) [inline web/app/scripts/src/directives/calculate-percentage.es.js]: Do we want to run a watch if question count is 0?

Comment (Minglee01) [inline web/app/scripts/src/directives/calculate-percentage.es.js]: Ok i get you

File: web/tests/calculate-percentage-directive-spec.js (hunk 2 above)
Comment (Minglee01) [inline web/tests/calculate-percentage-directive-spec.js]: I removed it as i didnt think it was that useful

Comment (Minglee01) [inline web/tests/calculate-percentage-directive-spec.js]: The test i added implys the same thing. I can readd

Comment (Minglee01) [inline web/tests/calculate-percentage-directive-spec.js]: The test i added implys the same thing. I can readd

**Comment by maxpeterson** [comment]
THis good to merge if you are done making changes @Minglee01

**Pull request merged**